### ISSUE TITLE
Spell-teaching item and out of battle healing conflict.

### ISSUE THREAD

**Issue by NathanWeasel** [ohrrpgce/ohrrpgce#1162]
I wanted to make the revive item in one of my games also teach one of the mages a revive spell, but the "teach spell" aspect seems to be overriding the targetability option for dead heroes. So outside of battle, the revive potion was refusing to target dead allies and trigger its attack...

So not only can I not teach the dead mage the revive spell this way, I also can't use the potion to revive other dead party members.

**Comment by rversteegen** [comment]
Yes, if an attack teaches a spell that overrides the out-of-battle attack or textbox. The plan is to replace the item file format (WIP) and make all three separate options (plus a 4th trigger script option); I think we should allow all of them to happen. If it both teaches and attacks, it makes sense that they are always targeted on the same hero. But the hero targetting UI will need improving to better indicate what's going on (that's the hard part).

We should preserve the current behaviour in existing games, but can do so without a backcompat bit. Instead, when upgrading to the new item data file format, just delete the OOB attack/textbox if the attack is set to teach a spell, since it does nothing.

**Comment by NathanWeasel** [comment]
For this particular project, I just removed the item teaching the spell and made the hero learn it by level-up instead. Being able to revive people out of battle with life potions was more important than what was basically an easter egg revive spell the mage would learn if you revived her that way.
